tests/unit/pnl/test_naive_pnl.py

```diff
@@ -159,6 +179,125 @@ def test_rebalancing_dates(self):
 
         self.assertTrue(no_months - 1 == len(unique_values_aud))
 
+    def test_make_pnl(self):
+
+        self.test_make_signal()
+
+        # Signal is produced daily. The calculation of the neutral level and standard
+        # deviation are also calculated daily. Only for a highly volatile asset would
+        # there be any value in calculating at a daily frequency. In most instances, the
+        # neutral level would remain fairly constant over the duration of a week.
+        dfw = self.signal_dfw
+
+        # The PnL DataFrame is appended to the instance DataFrame.
+        ret = 'EQXR'
+        sigs = ['CRY', 'GROWTH', 'INFL']
+        pnl = NaivePnL(self.dfd, ret=ret, sigs=sigs, cids=self.cids,
+                       start='2000-01-01', blacklist=self.blacklist,
+                       bms=["EUR_DUXR", "USD_DUXR"]
+                       )
+
+        pnl.make_pnl(sig='GROWTH', sig_op='zn_score_pan', rebal_freq='daily',
+                     vol_scale=None, rebal_slip=0, pnl_name='PNL_GROWTH',
+                     min_obs=252, iis=True, sequential=True, neutral='zero', thresh=None)
+
+        # Test the PnL value produced across the panel. Implicitly tests the PnL values
+        # for each cross-section.
+        # Confirms the category for the PnL is being named correctly.
+        pnl_df = pnl.df[pnl.df['xcat'] == 'PNL_GROWTH']
+
+        # Confirm the first PnL value for each cross-section is aligned to the first date
+        # of the respective signal, GROWTH. A PnL value should only be produced if the
+        # signal is available for the respective date.
+        expected_start = {'AUD': '2010-01-04', 'CAD': '2010-01-04', 'GBP': '2012-01-03',
+                          'NZD': '2010-01-04', 'USD': '2015-01-05', 'EUR': '2010-01-04'}
+
+        pnl_dfw = pnl_df.pivot(index='real_date', columns='cid',
+                               values='value')
+        cross_sections = self.cids
+        for c in cross_sections:
+            column = pnl_dfw.loc[:, c]
+            # Adjust the expected start dates by one day to account for the shift
+            # mechanism. The computed signal is used for the following day's position.
+            self.assertTrue(column.first_valid_index() ==
+                            pd.Timestamp(expected_start[c]) + pd.DateOffset(1))
+
+        # Choose a random date to confirm the logic of computing the PnL. Multiply each
+        # cross-section's signal by their respective return.
+        random_date = "2015-01-19"
```

Comment:
This is a fixed date, not a random one, and it would be safer to test the PnLs for a date range.

Reply:
Believe resolved.